cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -217,6 +218,29 @@ public void getArtifactInfo(HttpRequest request, HttpResponder responder,
   }
 
   @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/parents")
```

Comment:
Looks like the required data already exists in the ArtifactDetail object that the method fetches so it should be straightforward